File: java/cars-assemble/src/main/java/CarsAssemble.java
public class CarsAssemble {

    public double productionRatePerHour(int speed) {
        if (speed < 5) {
            return speed * 221;
        } else if (speed < 9) {
            return (speed * 221) * 0.9;
        } else if (speed == 9) {
            return 1989 * 0.8;
        } else {
            return 2210 * 0.77;
        }
    }

    public int workingItemsPerMinute(int speed) {
        return (int) productionRatePerHour(speed) / 60;
    }
}

File: fsharp/cars-assemble/CarsAssemble.fs
module CarsAssemble

let successRate (speed: int) : float =
    if speed = 0 then 0
    elif speed < 5 then 1
    elif speed < 9 then 0.9
    elif speed < 10 then 0.8
    else 0.77

let productionRatePerHour (speed: int) : float =
    float 221 * float speed * successRate speed

let workingItemsPerMinute (speed: int) : int =
    int (productionRatePerHour speed) / 60
